### DISCUSSION TITLE
Should we switch to Astro? (UPDATED)

### DISCUSSION THREAD

**Discussion by ykdojo** [ykdojo/defaang#109] [General]
Today I was reading a lot about Next.js, and in that process, I started learning about Astro.

It seems like Astro helps you make your website much more performant.

I'm actually now wondering if we should switch to Astro - which is a decision we should make soon while the project is in its earlier stage.

Any thoughts?

For more context:
- Astro has [server-side rendering](https://docs.astro.build/en/guides/server-side-rendering/), and it says it works with Vercel, so we can keep that part.
- The only thing that needs to be reactive on the website is going to be, I think, the upvoting/downvoting buttons for the questions. The other parts can be server-side rendered.
- The question I have is: when you compare a serverside rendered app in Next.js and Astro, will the performance be noticeably different?

UPDATE:

Based on @subhoghoshX and @iShibi's comments below, it seems like the right approach is to focus on the Next.js implementation right now.

If/when performance becomes an issue in the future, we can revisit it again.

**Comment by iShibi** [comment]
Astro is interesting and definitely an option to look into if we get performance issues with Next.js in the **"future"**. However currently, there are not enough reasons to use it. Next.js is battle-tested, tightly integrated with Vercel where defaang is hosted, and most of our current contributors seem to be well versed with it. Astro is new which means there are more chances of running into issues that have not been already fixed. Then there is also the issue of managing the global state in Astro.

Keeping all the things in mind I would suggest we let Astro become a little mature before making a move.

**Comment by ykdojo** [comment]
My conversation with @subhoghoshX on Discord:

![image](https://user-images.githubusercontent.com/1811651/185808187-e48c7693-5a6f-41fe-a3c0-2bde68b6dd64.png)
